File: output/gpt_4o/avatar/algo_based_translation/python/java/successful_translation/codeforces_61_B.java

import java.util.*;
import java.io.*;
import java.util.stream.Collectors;

public class codeforces_61_B {

    public static boolean valid(int i, int j, int n, int m) {
        return i >= 0 && i < n && j >= 0 && j < m;
    }

    public static double sumn(int i, int n) {
        return (n - i) * (i + n) / 2.0;
    }

    public static double sqfun(double a, double b, double c) {
        return (-b + Math.sqrt(b * b - 4 * a * c)) / (2 * a);
    }

    public static int[] value() throws IOException {
        return Arrays.stream(readLine().split("\\s+")).mapToInt(Integer::parseInt).toArray();
    }

    public static int[] values(BufferedReader reader) throws IOException {
        return Arrays.stream(reader.readLine().split("\\s+")).mapToInt(Integer::parseInt).toArray();
    }

    public static List<Integer> inlst() throws IOException {
        return Arrays.stream(readLine().split("\\s+")).map(Integer::parseInt).collect(Collectors.toList());
    }

    public static List<Integer> inlsts(BufferedReader reader) throws IOException {
        return Arrays.stream(reader.readLine().split("\\s+")).map(Integer::parseInt).collect(Collectors.toList());
    }

    public static int inp() throws IOException {
        return Integer.parseInt(readLine());
    }

    public static int inps(BufferedReader reader) throws IOException {
        return Integer.parseInt(reader.readLine());
    }

    public static String instr() throws IOException {
        return readLine();
    }

    public static List<String> stlst() throws IOException {
        return Arrays.asList(readLine().split("\\s+"));
    }

    public static String f(String s) {
        StringBuilder r = new StringBuilder();
        for (char i : s.toCharArray()) {
            if (i != '-' && i != ';' && i != '_') {
                r.append(Character.toLowerCase(i));
            }
        }
        return r.toString();
    }

    public static void solve() throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        List<String> l = new ArrayList<>();
        for (int i = 0; i < 3; i++) {
            l.add(f(reader.readLine()));
        }

        List<String> permutations = new ArrayList<>();
        permute(l, 0, permutations);

        int testCases = Integer.parseInt(reader.readLine());
        for (int i = 0; i < testCases; i++) {
            String input = f(reader.readLine());
            if (permutations.contains(input)) {
                System.out.println("ACC");
            } else {
                System.out.println("WA");
            }
        }
    }

    private static void permute(List<String> l, int index, List<String> result) {
        if (index == l.size()) {
            result.add(String.join("", l));
        } else {
            for (int i = index; i < l.size(); i++) {
                Collections.swap(l, i, index);
                permute(l, index + 1, result);
                Collections.swap(l, i, index);
            }
        }
    }

    public static void main(String[] args) throws IOException {
        solve();
    }

    private static String readLine() throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        return reader.readLine();
    }
}

File: dataset/avatar/Python/Code/codeforces_61_B.py
import collections
import heapq
import sys
import math
import itertools
import bisect
from io import BytesIO , IOBase
import os
def valid ( i , j , n , m ) :
    if i < n and i >= 0 and j >= 0 and j < m : return True
    return False
def sumn ( i , n ) :
    return ( n - i ) * ( i + n ) / 2
def sqfun ( a , b , c ) :
    return ( - b + math.sqrt ( b * b - 4 * a * c ) ) / 2 * a
def value ( ) : return tuple ( map ( int , input ( ).split ( ) ) )
def values ( ) : return tuple ( map ( int , sys.stdin.readline ( ).split ( ) ) )
def inlst ( ) : return [ int ( i ) for i in input ( ).split ( ) ]
def inlsts ( ) : return [ int ( i ) for i in sys.stdin.readline ( ).split ( ) ]
def inp ( ) : return int ( input ( ) )
def inps ( ) : return int ( sys.stdin.readline ( ) )
def instr ( ) : return input ( )
def stlst ( ) : return [ i for i in input ( ).split ( ) ]
def f ( s ) :
    r = ''
    for i in s :
        if i not in [ '-' , ';' , '_' ] : r += i.lower ( )
    return r
def solve ( ) :
    l = [ ]
    for i in range ( 3 ) :
        l.append ( f ( input ( ) ) )
    t = itertools.permutations ( l , len ( l ) )
    l = [ ]
    for i in t : l.append ( "".join ( i ) )
    for i in range ( inp ( ) ) :
        if f ( instr ( ) ) in l : print ( 'ACC' )
        else : print ( 'WA' )
if __name__ == "__main__" :
    solve ( )

